src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/PaymentAccountServiceImpl.java

```diff
@@ -61,4 +69,63 @@ public Organisation findPaymentAccountsByEmail(String email) {
         }
         return organisation;
     }
-}
+
+    @Transactional
+    public void deleteUserAccountMaps(Organisation organisation) {
+        ProfessionalUser user = organisation.getUsers().get(0).toProfessionalUser();
+        List<PaymentAccount> paymentAccount = organisation.getPaymentAccounts();
+        List<UserAccountMapId> accountsToDelete = generateListOfAccountsToDelete(user, paymentAccount);
+
+        userAccountMapRepository.deleteByUserAccountMapIdIn(accountsToDelete);
+    }
+
+    @Transactional
+    public void deletePaymentAccountsFromOrganisation(Organisation organisation) {
+        List<UUID> accountIds = new ArrayList<>();
+
+        organisation.getPaymentAccounts().forEach(account -> accountIds.add(account.getId()));
+
+        paymentAccountRepository.deleteByIdIn(accountIds);
+
+        //The below lines are required to set the Organisation's Payment Accounts List to empty
+        //if this is not done, the Organisation's list will still contain the previous Accounts
+        List<PaymentAccount> resetOrganisationPaymentAccounts = new ArrayList<>();
+        organisation.setPaymentAccounts(resetOrganisationPaymentAccounts);
+    }
+
+    public void addPaymentAccountsToOrganisation(PbaEditRequest pbaEditRequest, Organisation organisation) {
+        organisationServiceImpl.addPbaAccountToOrganisation(pbaEditRequest.getPaymentAccounts(), organisation);
+    }
+
+    public PbaResponse addUserAndPaymentAccountsToUserAccountMap(Organisation organisation) {
+        List<PaymentAccount> paymentAccounts = organisation.getPaymentAccounts();
+        SuperUser superUser = organisation.getUsers().get(0);
+
+        organisationServiceImpl.persistedUserAccountMap(superUser.toProfessionalUser(), paymentAccounts);
+
+        return new PbaResponse(HttpStatus.OK.toString(), HttpStatus.OK.getReasonPhrase());
+    }
+
+    public List<UserAccountMapId> generateListOfAccountsToDelete(ProfessionalUser user, List<PaymentAccount> accounts) {
+        List<UserAccountMapId> userAccountMapIds = new ArrayList<>();
+
+        accounts.forEach(account -> {
+            if (null != user && null != account) {
+                UserAccountMapId userAccountMapId = new UserAccountMapId(user, account);
+                userAccountMapIds.add(userAccountMapId);
+            }
+        });
+        return userAccountMapIds;
+    }
+
+    public void validatePaymentAccounts(List<String> paymentAccounts) {
+        if (paymentAccounts != null) {
+
+            paymentAccounts.forEach(pbaAccount -> {
+                if (pbaAccount == null || !pbaAccount.matches("(PBA|pba).*") || !pbaAccount.matches("^[a-zA-Z0-9]+$")) {

```

Comment:
As discussed above. This should be changed to handle multiple pbas and use single-responsibility. However you can keep this method and just change its guts.